core/src/test/java/com/dtolabs/rundeck/core/common/TestNodesYamlParser.java

```diff
@@ -307,24 +307,25 @@ public void testParseInvalid_allow_missing_hostname() throws Exception {
         Assert.assertEquals(null, recv.nodes.get("bill").getHostname());
 
     }
-    public void testParseInvalid_unexpecteddatatype() throws Exception {
-            //unexpected data type for string field
-            testReceiver recv = new testReceiver();
-            ByteArrayInputStream is = new ByteArrayInputStream(
-                ("test: \n"
-                 + "  nodename: bill\n"
-                 + "  hostname: test\n"
-                 + "  username: {test:value, a:b}\n").getBytes());
-
-            NodesYamlParser nodesYamlParser = new NodesYamlParser(is, recv);
-            try {
-                nodesYamlParser.parse();
-                fail("parsing should fail: " + recv.nodes);
-            } catch (NodeFileParserException e) {
-                assertTrue(e.getCause() instanceof YAMLException);
-            }
-
-        }
+// AS OF SNAKEYAML 1.21 scalars can now contain colons, so this test will no longer fail
+//    public void testParseInvalid_unexpecteddatatype() throws Exception {
+//            //unexpected data type for string field
+//            testReceiver recv = new testReceiver();
+//            ByteArrayInputStream is = new ByteArrayInputStream(
+//                ("test: \n"
+//                 + "  nodename: bill\n"
+//                 + "  hostname: test\n"
+//                 + "  username: {test:value, a:b}\n").getBytes());
```

Comment:
can this be changed to valid yaml array syntax, so that the test still tries to parse invalid datatype for this field?

Reply:
otherwise, could this just be rewritten instead of removed?  i'd have to review exactly what it is testing, but it appears to try to send a non-string attribute value to the node yaml parser, expecting an error. if the current input evaluates to a valid string, perhaps just changing it to result in a map/array would work

Reply:
Added the test back